tests/api/testcl.hpp

```diff
@@ -757,3 +757,20 @@ class WithProfiledCommandQueue : public WithCommandQueue {
 protected:
     void SetUp() override { SetUpQueue(CL_QUEUE_PROFILING_ENABLE); }
 };
+
+class WithPrintfEnabled : public WithCommandQueue {
+protected:
+    void SetUp() override{};
+    void TearDown() override {
+        cl_int err = clReleaseContext(m_context);
+        ASSERT_CL_SUCCESS(err);
```

Comment:
We will need to call `WithContext::TearDown()` and `WithCommandQueue::TearDown()` depending on whether the creation succeeded.